File: 2432-the-employee-that-worked-on-the-longest-task/2432-the-employee-that-worked-on-the-longest-task.cpp
class Solution {
public:
    int hardestWorker(int n, vector<vector<int>>& logs) {
        int id=logs[0][0];
        int max_val=logs[0][1];
        
        for(int i=1;i<logs.size();i++){
            if((logs[i][1]-logs[i-1][1])>max_val){
                max_val=logs[i][1]-logs[i-1][1];
                id=logs[i][0];
            }
            else if((logs[i][1]-logs[i-1][1])==max_val){
                max_val=logs[i][1]-logs[i-1][1];
                id=min(id,logs[i][0]);
            }
        }
        
        return id;
    }
};

File: 2432-the-employee-that-worked-on-the-longest-task/2432-the-employee-that-worked-on-the-longest-task.py
class Solution:
    def hardestWorker(self, n: int, logs: List[List[int]]) -> int:
        id=logs[0][0]
        max_val=logs[0][1]
        
        for i in range(1,len(logs)):
            if (logs[i][1]-logs[i-1][1])>max_val:
                max_val=logs[i][1]-logs[i-1][1]
                id=logs[i][0]
            elif (logs[i][1]-logs[i-1][1])==max_val:
                max_val=logs[i][1]-logs[i-1][1]
                id=min(id,logs[i][0])
        return id